homeassistant/components/tplink/__init__.py

```diff
@@ -3,36 +3,58 @@
 
 import asyncio
 from datetime import timedelta
-from typing import Any
-
-from kasa import SmartDevice, SmartDeviceException
-from kasa.discover import Discover
+import logging
+from typing import Any, Final, Optional
+
+from kasa import (
+    AuthenticationException,
+    Credentials,
+    DeviceConfig,
+    Discover,
+    SmartDevice,
+    SmartDeviceException,
+)
 
 from homeassistant import config_entries
 from homeassistant.components import network
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import (
+    CONF_AUTHENTICATION,
     CONF_HOST,
     CONF_MAC,
     CONF_NAME,
+    CONF_PASSWORD,
+    CONF_USERNAME,
     EVENT_HOMEASSISTANT_STARTED,
 )
 from homeassistant.core import HomeAssistant, callback
-from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
 from homeassistant.helpers import (
     config_validation as cv,
     device_registry as dr,
     discovery_flow,
 )
 from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.httpx_client import create_async_httpx_client
 from homeassistant.helpers.typing import ConfigType
 
-from .const import DOMAIN, PLATFORMS
+from .const import (
+    CONF_DEVICE_CONFIG,
+    CONNECT_TIMEOUT,
+    DISCOVERY_TIMEOUT,
+    DOMAIN,
+    PLATFORMS,
+)
 from .coordinator import TPLinkDataUpdateCoordinator
 
-DISCOVERY_INTERVAL = timedelta(minutes=15)
+# DISCOVERY_INTERVAL = timedelta(minutes=15)
+DISCOVERY_INTERVAL = timedelta(seconds=30)
```

Comment:
Hopefully this is for debugging as 30s is far too often to send broadcast traffic

Reply:
It was for debugging, I've reverted it to 15 mins